### PR TITLE
Model evanphx/json-patch

### PR DESCRIPTION
**Pull request by owen-mc**
I need to run a dist-compare before merging, but I thought I'd get review feedback first

### CODE REVIEW

**Review by smowton — APPROVED** [review]
Have you checked the context in the target CVE to verify you're modelling the relevant function there?

**Comment by owen-mc** [comment]
Yes. The target CVE requires a model for `CreateMergePatch`. The package was small enough that I just modelled everything in the most recent release.

File: ql/src/semmle/go/frameworks/EvanphxJsonPatch.qll [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      inp.isParameter(0) and
+      outp.isResult(0)
+    }
+  }
+
+  private class Apply extends TaintTracking::FunctionModel {
+    Apply() {
+      this
+          .hasQualifiedName(packagePath(),
+            ["Apply", "ApplyWithOptions", "ApplyIndent", "ApplyIndentWithOptions"])
```
Comment (sauyon) [inline ql/src/semmle/go/frameworks/EvanphxJsonPatch.qll:51]:
```suggestion
      exists(string fn |
        fn = ["Apply", "ApplyWithOptions", "ApplyIndent", "ApplyIndentWithOptions"]
      |
        this.hasQualifiedName(packagePath(), fn)
      )
```

File: ql/test/library-tests/semmle/go/frameworks/EvanphxJsonPatch/TaintFlows.ql [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import go
```
Comment (sauyon) [inline ql/test/library-tests/semmle/go/frameworks/EvanphxJsonPatch/TaintFlows.ql:1]: Could you just use the standard data-flow test from `ql/test/library-tests/semmle/go/dataflow/TypeAssertions/DataFlow.ql` and adapt it for taint flow?

**Review by sauyon — COMMENTED** [review]
One stylistic comment.

**Pull request merged**